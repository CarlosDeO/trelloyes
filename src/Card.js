import React from 'react';
import './Card.css';

class Card extends React.Component {
    render() {
        console.log(this.props);
        return ( 
            <div className = "Card" >
            <button 
            type ='button'  
            onClick={() => this.props.onClickDelete(this.props.id)}> 
            Delete
            </button> 
            <h3> {this.props.title} </h3> 
            <p> {this.props.content} </p> 
            </div>
        )
    }
}

Card.propTypes = {
    onClickDelete: () => {}
  }

export default Card;