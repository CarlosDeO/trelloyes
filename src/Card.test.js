import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import renderer from 'react-test-renderer';


describe('Card component', () => {

   const allCards = {
        'a': { id: 'a', title: 'First card', content: 'lorem ipsum' },
        'b': { id: 'b', title: 'Second card', content: 'lorem ipsum' },
        'c': { id: 'c', title: 'Third card', content: 'lorem ipsum' },
        'd': { id: 'd', title: 'Fourth card', content: 'lorem ipsum' },
        'e': { id: 'e', title: 'Fifth card', content: 'lorem ipsum' },
        'f': { id: 'f', title: 'Sixth card', content: 'lorem ipsum' },
        'g': { id: 'g', title: 'Seventh card', content: 'lorem ipsum' },
        'h': { id: 'h', title: 'Eighth card', content: 'lorem ipsum' },
        'i': { id: 'i', title: 'Ninth card', content: 'lorem ipsum' },
        'j': { id: 'j', title: 'Tenth card', content: 'lorem ipsum' },
        'k': { id: 'k', title: 'Eleventh card', content: 'lorem ipsum' },
        'l': { id: 'l', title: 'Twelfth card', content: 'lorem ipsum' },
        'm': { id: 'm', title: 'Thirteenth card', content: 'lorem ipsum' },
      }
    

    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<Card />, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('render the UI as expected', () => {
        const tree = renderer
        .create(<Card
            key={allCards['a'].id}
            title={allCards['a'].title}
            content={allCards['a'].content}
          />)
          .toJSON();
          expect(tree).toMatchSnapshot();
    });
});