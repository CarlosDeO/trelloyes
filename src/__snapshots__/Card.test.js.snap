// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component render the UI as expected 1`] = `
<div
  className="Card"
>
  <button
    onClick={[Function]}
    type="button"
  >
    Delete
  </button>
  <h3>
     
    First card
     
  </h3>
  <p>
     
    lorem ipsum
     
  </p>
</div>
`;
